test/test_dataloader.py

```diff
@@ -8,15 +8,26 @@
 import time
 import traceback
 import unittest
+import subprocess
 from torch import multiprocessing
 from torch.utils.data import Dataset, TensorDataset, DataLoader, ConcatDataset
 from torch.utils.data.dataset import random_split
-from torch.utils.data.dataloader import default_collate, ExceptionWrapper
+from torch.utils.data.dataloader import default_collate, ExceptionWrapper, MANAGER_STATUS_CHECK_INTERVAL
 from common import TestCase, run_tests, TEST_NUMPY, IS_WINDOWS
-from common_nn import TEST_CUDA
 
+TEST_CUDA = torch.cuda.is_available()
 
-JOIN_TIMEOUT = 17.0 if IS_WINDOWS else 4.5
+# We need spawn start method for test_manager_unclean_exit
+if sys.version_info[0] == 3:
```

Comment:
Why is the version test necessary here?

Reply:
`multiprocessing` doesn't have `set_start_method()` on Python 2.7 and can only use the `fork` start method on Linux. I can add a comment to explain it.